### ISSUE TITLE
[BUG] APM Profiler causes app crashes

### ISSUE THREAD

**Issue by JanKnipp** [elastic/apm-agent-dotnet#1918]
Hi,

we've noticed a strange issue with the profiler which is currently leading to crashing applications/app pools/powershell, ...

I've used the powershell example as it is showing a nice error message as an example

### APM Agent version
APM Profiler 1.18

### Environment

Windows Server 2021 R2 / Windows Server 2016

### Describe the bug
When trying to open a new powershell session we are agreed with the following error:
![grafik](https://user-images.githubusercontent.com/10583181/203347153-7a8c5fe5-15a3-4142-af1b-efbcb082f5b0.png)
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 5, kind: PermissionDenied, message: "Access is denied." }', src\elastic_apm_profiler\src\profiler\env.rs:371:18
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: Rust panics must be rethrown

![grafik](https://user-images.githubusercontent.com/10583181/203348113-c698965d-a967-4a8b-a715-054273bad209.png)
Powershell is running with PID56608 which shows an existing profiler log file

![grafik](https://user-images.githubusercontent.com/10583181/203348548-d7a94509-bd8f-4342-bccc-949ff63912c7.png)

![grafik](https://user-images.githubusercontent.com/10583181/203350542-643d1083-e5ef-44f8-873a-c4d0abeb33e3.png)

No other powershell process is (currently) running and of course no one with an equal PID.

We're starting a few scripts which create quite a lot of profiler logs (log dir current holds 10970 powershell profiler logs).

<speculation>
My guess is that in these cases we are dealing with an ownership issue. The sample above was created by a scheduled task running the script with administrative privileges and so ownership was limited to the Administrators group with read access for Users.
Since we're creating a lot of powershell shell profiler logs we have a high number of PIDs used with specific users.
When a different user with different privileges tries to open powershell the rust codes tries to append data to that file which in certain cases fails as the user does not have correct permissions for the file and might not be able to append.

This can be an issue for all processes where owners might change or where the process itself can be used by different users (powershell, w3c.exe, nssm (non sucking service manager), ...)
</speculation>

Code crashes in the following line:
https://github.com/elastic/apm-agent-dotnet/blob/7c8adc295ba64b4aceaf271e496220571f7f5fb0/src/elastic_apm_profiler/src/profiler/env.rs#L367
`let file = RollingFileAppender::builder()
                .append(true)
                .encoder(Box::new(pattern))
                .build(&log_file_name, Box::new(policy))
                .unwrap();`

#### Expected behavior
My expectation would be that even if the profiler cannot create a log file it should never lead to an application crash, so log file creation should be "safe". I'd rather take no log file than a crashed application

#### Actual behavior
Profiler crashes application

**Comment by z1c0** [comment]
Hi @JanKnipp,

I 100% agree that the profiler must not bring down an application for whatever reason and we will look into the issue at our next convenience.

I'm curious though: was it your actual intention to monitor a PowerShell process or is this an unwanted side-effect of a system-wide configuration (profiler environment variables)?
In the latter case, I suggest narrowing down the environment variables to the application pool in interest (I just assumed an IIS use case here). Is that an option for you or do you have a different use case?

**Comment by JanKnipp** [comment]
Hi @z1c0,
you are right. In this case it was not our intent to monitor powershell but a global configuration of the profiler settings which was necessary due to using Windows Server 2012 or to be more specific an old IIS version that does not allow app-pool specific environments.

It happened on a test system that also runs multiple IIS sites and we just looked at the machine because we saw crashing app-pools due to the elastic profiler. The sites are configured with different users and I think this also might have caused the pool crashes as this will be logged as
 `elastic_apm_profiler_w3wp_%PID%.log`

but in this case it is not as clear as in the powershell example as this shows a "nice" error message which is easier to understand as the rather generic event log entry for the app-pool

```
Faulting application name: w3wp.exe, version: 10.0.14393.0, time stamp: 0x57899b8a
Faulting module name: elastic_apm_profiler.dll, version: 0.0.0.0, time stamp: 0x63486392
Exception code: 0xc0000409
Fault offset: 0x00000000000ed7e1
Faulting process id: 0x3044
Faulting application start time: 0x01d8fd7636a0b3ba
Faulting application path: c:\windows\system32\inetsrv\w3wp.exe
```

We also did use the env variable to exclude powershell
`ELASTIC_APM_PROFILER_EXCLUDE_PROCESSES=powershell.exe;..`
but this did not include the log file creation.
